pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -433,14 +433,10 @@ func (c *InstallerController) manageInstallationPods(ctx context.Context, operat
 
 		klog.Infof("%s and needs new revision %d", nodeChoiceReason, revisionToStart)
 
-		newCurrNodeState := currNodeState.DeepCopy()
-		newCurrNodeState.TargetRevision = revisionToStart
-		if newCurrNodeState.LastFailedRevision != revisionToStart {
-			newCurrNodeState.LastFailedRevisionErrors = nil
-			newCurrNodeState.LastFailedCount = 0
-			newCurrNodeState.LastFailedTime = nil
-		} else if newCurrNodeState.LastFailedCount == 0 {
-			newCurrNodeState.LastFailedCount = 1
+		newCurrNodeState := &operatorv1.NodeStatus{
```

Comment:
is it okay to clean `LastFailedRevisionErrors` because a failed revision can never reach this code?

Reply:
The revision is a new one. We already wiped out all lastFailed* fields before, part of the API.